File: frontend/src/types/checkin.ts
export interface CheckinResponse {
  id: number;
  student_id: number;
  student_name?: string;
  checkin_date: string;
  content?: string;
  created_at: string;
}

export interface CheckinStats {
  student_id: number;
  student_name: string;
  total_days: number;
  consecutive_days: number;
  checkin_dates: string[];
}

export interface CheckinRankItem {
  student_id: number;
  student_name: string;
  total_days: number;
  consecutive_days: number;
}

File: backend/app/schemas/checkin.py
from pydantic import BaseModel
from typing import Optional
from datetime import date, datetime


class CheckinCreate(BaseModel):
    student_id: int
    checkin_date: Optional[date] = None  # 默认今天
    content: Optional[str] = None


class CheckinResponse(BaseModel):
    id: int
    student_id: int
    student_name: Optional[str] = None
    checkin_date: date
    content: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class CheckinStatsResponse(BaseModel):
    student_id: int
    student_name: str
    total_days: int
    consecutive_days: int
    checkin_dates: list[date]


class CheckinRankItem(BaseModel):
    student_id: int
    student_name: str
    total_days: int
    consecutive_days: int


class CheckinRankResponse(BaseModel):
    items: list[CheckinRankItem]
